runtime/parser/function.go

```diff
@@ -167,11 +167,34 @@ func parseParameter(p *parser) (*ast.Parameter, error) {
 		return nil, err
 	}
 
+	p.skipSpaceAndComments()
+
+	var defaultArgument ast.Expression
+
+	if expectDefaultArgument {
+		if !p.current.Is(lexer.TokenEqual) {
+			return nil, p.syntaxError(
+				"expected a default argument after type annotation, got %s",
+				p.current.Type,
+			)
+		}
+
+		// Skip the =
+		p.nextSemanticToken()
+
+		defaultArgument, err = parseExpression(p, lowestBindingPower)
+		if err != nil {
+			return nil, err
+		}
+
+	}
```

Comment:
Maybe add an else-case here, which still checks for an equal token and reports a helpful error, e.g. "unexpected default argument". This would improve the DX a bit over the current, generic error `"expected comma or end of parameter list, got '='"`